test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -1208,22 +1213,27 @@ public override void Inject<T>(PropagationContext context, T carrier, Action<T,
         }
     }
 
-    private class TestSampler(SamplingDecision samplingDecision, IEnumerable<KeyValuePair<string, object>> attributes = null) : Sampler
+    private class TestSampler(SamplingDecision samplingDecision, IEnumerable<KeyValuePair<string, object>>? attributes = null) : Sampler
     {
         private readonly SamplingDecision samplingDecision = samplingDecision;
-        private readonly IEnumerable<KeyValuePair<string, object>> attributes = attributes;
+        private readonly IEnumerable<KeyValuePair<string, object>>? attributes = attributes;
 
         public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
         {
-            return new SamplingResult(this.samplingDecision, this.attributes);
+            if (this.attributes != null)
+            {
+                return new SamplingResult(this.samplingDecision, this.attributes);
```

Comment:
@open-telemetry/dotnet-approvers, I have a question about accepted parameters here.

Attributes is forced to be not null https://github.com/open-telemetry/opentelemetry-dotnet/blob/core-1.8.1/src/OpenTelemetry/Trace/SamplingResult.cs#L35, but the constructor only with SamplingDecision as parameters, internally [is calling this one with a `null` as a value](https://github.com/open-telemetry/opentelemetry-dotnet/blob/core-1.8.1/src/OpenTelemetry/Trace/SamplingResult.cs#L15-L16).

Is it intentional? Or it can be changed?  Similar case is for `traceStateString` parameter.

Reply:
Does not look intentional. @CodeBlanch - do you know?

Reply:
Created https://github.com/open-telemetry/opentelemetry-dotnet/pull/5614 to discuss it/fix it.